File: src/main/java/com/softwareplace/security/service/AuthorizationUserService.java
package com.softwareplace.security.service;

import java.util.HashMap;
import java.util.Map;

import javax.servlet.FilterChain;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

import org.springframework.security.core.Authentication;

import com.softwareplace.security.model.RequestUser;
import com.softwareplace.security.model.UserData;

public interface AuthorizationUserService {

	UserData userData(RequestUser user);

	UserData userData(String username);

	long expirationTime();

	String secret();

	default void successfulAuthentication(HttpServletRequest request, HttpServletResponse response, FilterChain chain, Authentication authResult) {

	}

	default Map<String, Object> claims(HttpServletRequest httpServletRequest, UserData userData) {
		return new HashMap<>();
	}
}

File: app/src/main/kotlin/com/softwareplace/springsecurity/service/AuthorizationUserService.kt
package com.softwareplace.springsecurity.service

import com.softwareplace.springsecurity.model.RequestUser
import com.softwareplace.springsecurity.model.UserData
import jakarta.servlet.FilterChain
import jakarta.servlet.http.HttpServletRequest
import jakarta.servlet.http.HttpServletResponse
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken
import org.springframework.security.core.Authentication
import org.springframework.security.core.userdetails.UserDetailsService

interface AuthorizationUserService : UserDetailsService {

    fun findUser(user: RequestUser): UserData?
    fun findUser(authToken: String): UserData?

    fun successfulAuthentication(
        request: HttpServletRequest,
        response: HttpServletResponse,
        chain: FilterChain,
        authResult: Authentication
    ) {
        // ignored
    }

    fun claims(httpServletRequest: HttpServletRequest, userData: UserData): Map<String, List<Any>> {
        return HashMap()
    }

    fun authenticate(
        request: HttpServletRequest,
        defaultHandler: (request: HttpServletRequest) -> UsernamePasswordAuthenticationToken
    ): UsernamePasswordAuthenticationToken {
        return defaultHandler.invoke(request)
    }
}
